FrameworkSystem/scripts/dirac-populate-component-db.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python
+
+"""
+Populates the database with the current installations of components
+Requires the local dirac.cfg to contain the full list of host 
+under Registry/Hosts
+"""
+
+__RCSID__ = "$Id$"
+
+import sys
+from datetime import datetime
+from DIRAC.ConfigurationSystem.Client.CSAPI import CSAPI
+from DIRAC.Core.Base import Script
+from DIRAC.FrameworkSystem.Client.SystemAdministratorIntegrator \
+  import SystemAdministratorIntegrator
+from DIRAC.FrameworkSystem.Client.ComponentMonitoringClient \
+  import ComponentMonitoringClient
+
+Script.setUsageMessage( '\n'.join( [ __doc__.split( '\n' )[1],
+                        'Usage:',
+                        '  %s [option|cfgfile] ... [debug]' % Script.scriptName,
+                        'Arguments:',
+                        '  debug:  Debug mode on/off'] ) )
+
+Script.parseCommandLine()
+args = Script.getPositionalArgs()
+
+if len( args ) > 0 and args[0] == 'debug':
+  debug = True
+else:
+  debug = False
+
+componentType = ''
+
+# Retrieve information from all the hosts
+client = SystemAdministratorIntegrator()
+resultAll = client.getOverallStatus()
+resultHosts = client.getHostInfo()
+resultInfo = client.getInfo()
+resultAllDB = client.getDatabases()
+resultAvailableDB = client.getAvailableDatabases()
+
+if not resultAll[ 'OK' ]:
```

Comment:
And you can then use DIRAC.exit. It avoids to indent all your code in the else clause